### PR TITLE
[corda-ent] Added the helm chart and dockerfile for CENM Zone service

### PR DESCRIPTION
**Pull request by surabhi-dudhediya**
Signed-off-by: surabhi-dudhediya <surabhi.dudhediya@accenture.com>

**Changelog**
- Add helm chart and dockerfile for CENM Zone service

**Reviewed by**
@suvajit-sarkar @jagpreetsinghsasan @sownak 

 

**Linked issue**
#1735

### CODE REVIEW

File: platforms/r3-corda-ent/images/enterprise-zone1.5.1.dockerfile [hunk 1]
```diff
@@ -0,0 +1 @@
+FROM corda/enterprise-zone:1.5.1-zulu-openjdk8u242
```
Comment (suvajit-sarkar) [inline platforms/r3-corda-ent/images/enterprise-zone1.5.1.dockerfile:1]: newline

Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/images/enterprise-zone1.5.1.dockerfile]: fixed

File: platforms/r3-corda-ent/charts/zone/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: init-certificates
+          image: {{ .Values.image.initContainerName }}
+          imagePullPolicy: Always
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/zone/templates/deployment.yaml:41]: dont use Always for initcontainer

File: platforms/r3-corda-ent/charts/zone/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{{- define "labels.custom" -}}
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/zone/templates/_helpers.tpl:1]: labels.custom has not been used. Ideally use this in all templates

File: platforms/r3-corda-ent/charts/zone/values.yaml [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+#############################################################
+#                   Basic Configuration                     #
+#############################################################
+# Provide the name of the deployment
+# Eg. nodeName: zone
+nodeName: 
+
+# Display more messages while executing Kubernetes deployment
+bashDebug: false
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/zone/values.yaml:19]: This key has never been used. Please check all keys in values.yaml are used in templates. or delete them if not needed

File: platforms/r3-corda-ent/charts/zone/values.yaml [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+  # Eg. password: "example-db-password"
+  password:
+  # Migrations of database can be run as part of the startup of Zone, if set to true. 
+  # If set to false, it will be run prior to setting up the Zone.
+  # Eg. runMigration: "true"
+  runMigration:
+
+# Volume size for h2/ directory
+# Eg. volumeSizeZoneH2: 1Gi
+volumeSizeZoneH2:
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/zone/values.yaml:102]: define volumes in a single place, not multiple times

File: platforms/r3-corda-ent/charts/zone/values.yaml [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+  # Provide the vault role used.
+  # Eg. role: vault-role
+  role:
+  # Provide the authpath configured to be used.
+  # Eg. authpath: entcordacenm
+  authPath:
+  # Provide the service account name autheticated to vault.
+  # NOTE: Make sure that the service account is already created and autheticated to use the vault.
+  # Eg. serviceaccountname: vault-auth
+  serviceAccountName:
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/zone/values.yaml:222]: duplicate use of service account name

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/r3-corda-ent/charts/zone/templates/deployment.yaml (hunk 2 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/templates/deployment.yaml]: fixed

File: platforms/r3-corda-ent/charts/zone/templates/_helpers.tpl (hunk 3 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/templates/_helpers.tpl:1]: fixed

File: platforms/r3-corda-ent/charts/zone/values.yaml (hunk 4 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/values.yaml]: fixed

File: platforms/r3-corda-ent/charts/zone/values.yaml (hunk 5 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/values.yaml]: fixed

File: platforms/r3-corda-ent/charts/zone/values.yaml (hunk 6 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/values.yaml:215]: fixed

File: platforms/r3-corda-ent/charts/zone/values.yaml [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+# Name of Kubernetes secret storing credentials to Docker registry
+# Provide the docker-registry secret created and stored in kubernetes cluster as a secret.
+imagePullSecrets:
+  # Eg. name: regcred
+  - name:
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  # Eg. create: false
+  create:
```
Comment (sownak) [inline platforms/r3-corda-ent/charts/zone/values.yaml:148]: Don't think this create is used anywhere

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/r3-corda-ent/charts/zone/values.yaml (hunk 7 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/values.yaml]: fixed

File: platforms/r3-corda-ent/charts/zone/values.yaml [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+        # Provide the requests memory for node
+        # Eg. memory: 1Gi
+        memory:
+
+  zoneJar:
+    # The directory where the Zone Service .jar file is stored
+    # Eg. path: bin
+    path:
+
+  podSecurityContext:
```
Comment (jagpreetsinghsasan) [inline platforms/r3-corda-ent/charts/zone/values.yaml:81]: These can be removed and hard coded straight in charts

File: platforms/r3-corda-ent/charts/zone/values.yaml [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+    # Eg. volumeSizeZoneH2: 1Gi
+    volumeSizeZoneH2:
+    # Volume size for logs/ directory
+    # Eg. volumeSizeZoneLogs: 5Gi
+    volumeSizeZoneLogs:
+    # Volume size for data/ directory
+    # Eg. volumeSizeZoneData: 1Gi
+    volumeSizeZoneData:
+
+  pod:
```
Comment (jagpreetsinghsasan) [inline platforms/r3-corda-ent/charts/zone/values.yaml:64]: These can be removed and hard coded straight in the charts

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

File: platforms/r3-corda-ent/charts/zone/values.yaml (hunk 8 above)
Comment (surabhi-dudhediya) [reply platforms/r3-corda-ent/charts/zone/values.yaml]: fixed

**Review by sownak — APPROVED** [review]

**Pull request merged**